### PR TITLE
Plugin doc migration from wiki to github

### PR DESCRIPTION
**Pull request by deeepsig**
- Modified some content in the README.md file using https://github.com/jenkins-infra/plugins-wiki-docs/tree/master/mattermost
- Modified grammar to enhance readability.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
   <version>3.1.2-SNAPSHOT</version>
   <name>Mattermost Notification Plugin</name>
   <description>A Build status publisher that notifies channels on a Mattermost server</description>
-  <url>https://wiki.jenkins-ci.org/display/JENKINS/Mattermost+Plugin</url>
+  <url>https://plugins.jenkins.io/mattermost/</url>
```
Comment (halkeye) [inline pom.xml]: this url should be github, its where the source of the readme data comes from.

File: README.md [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
-2. Configure an incoming webhook
+1. The first step is to set up a Mattermost server
+2. Then Configure an incoming webhook
 3. Install this plugin on your Jenkins server
 4. **Add it as a Post-build action** in your Jenkins job.
 
 # Developer instructions
 
-Install Maven and JDK.
+Make sure you have Maven and JDK installed.
```
Comment (StackScribe) [inline README.md:29]:
```suggestion
Make sure that Maven and a JDK are installed.
```

**Review by StackScribe — COMMENTED** [review]
Nicely done.  Just a couple optional comments.

File: pom.xml (hunk 1 above)
Comment (deeepsig) [reply pom.xml]: Thanks for reviewing @halkeye !. I shall change it

**Comment by deeepsig** [comment]
> Nicely done. Just a couple optional comments.

Thanks for the kind words @StackScribe :). I shall take a look into them.

File: README.md (hunk 2 above)
Comment (deeepsig) [reply README.md:29]: @StackScribe , How about we use : 
`Make sure you have Maven and a JDK installed.` 
sounds good?

File: pom.xml (hunk 1 above)
Comment (deeepsig) [reply pom.xml]: Done 👍🏽

**Comment by cinlloc** [comment]
@halkeye @StackScribe would it be possible to merge this one? Is there work still to achieve on it?

**Comment by halkeye** [comment]
Only the maintainer can merge. My guess is that it's @jovandeginste 

Otherwise a new maintainer would need to apply to take over

**Comment by covert8** [comment]
LGTM

**Pull request merged**